src/ErrorSparseTask.cpp

```diff
@@ -23,28 +24,35 @@ ErrorSparseTask::ErrorSparseTask(uint64_t model_size,
                                  uint64_t features_per_sample,
                                  uint64_t nworkers,
                                  uint64_t worker_id,
-                                 const std::string& ps_ip,
-                                 uint64_t ps_port)
+                                 const Configuration& config,
+                                 const std::vector<std::string>& ps_ips,
+                                 const std::vector<uint64_t>& ps_ports)
     : MLTask(model_size,
              batch_size,
              samples_per_batch,
              features_per_sample,
              nworkers,
              worker_id,
-             ps_ip,
-             ps_port) {
-  ps_port = ps_port;
+             config,
+             ps_ips,
+             ps_ports) {
+  std::atomic_init(&curr_error, 0.0);
+  port_num = ps_ports.at(ps_ports.size() - 1) + 1;
```

Comment:
Are we making any assumption about this ps_ports (e.g., they are all contiguous)? If so, we should make that clear somewhere with comments

Reply:
Done